api4/config.go

```diff
@@ -260,6 +265,15 @@ func patchConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Do not allow marketplace URL to be toggled if plugin uploads are disabled.
+	if cfg.PluginSettings.MarketplaceURL != nil && cfg.PluginSettings.EnableUploads != nil {
+		// Breaking it down to 2 conditions to make it simple.
+		if *cfg.PluginSettings.MarketplaceURL != *appCfg.PluginSettings.MarketplaceURL && !*cfg.PluginSettings.EnableUploads {
```

Comment:
Shouldn't we check the `appCfg.PluginSettings.EnableUploads` as it is the source of truth?
```suggestion
		if *cfg.PluginSettings.MarketplaceURL != *appCfg.PluginSettings.MarketplaceURL && !*appCfg.PluginSettings.EnableUploads {
```

Reply:
The value already gets replaced in L143, but you are right. Using `appCfg` makes it clear.